src/pages/Order/components/Form/Form.tsx

```diff
@@ -0,0 +1,207 @@
+import { TargetLanguageField } from './components/TargetLanguageField'
+import { schema, targetLanguageSchema } from './schema'
+import type { Language, Voice, VoiceType } from './types'
+
+export const formProps = {
+  url: {
+    type: String,
+    required: true,
+  },
+  duration: {
+    type: String,
+    required: true,
+  },
+  sourceLanguages: {
+    type: Object as PropType<Language[]>,
+    required: true,
+  },
+  targetLanguages: {
+    type: Object as PropType<Language[]>,
+    required: true,
+  },
+  voices: {
+    type: Array as PropType<Voice[]>,
+    required: true,
+  },
+  onSubmit: {
+    type: Function as PropType<(data: TypeOf<typeof schema>) => void>,
+  },
+} as const
+
+export type FormProps = ExtractPropTypes<typeof formProps>
+
+export const Form = defineComponent({
+  name: 'Form',
+  props: formProps,
+  setup(props) {
+    const getId = () => Date.now()
+    const { t } = useI18n()
+
+    const form = useForm({
+      validationSchema: schema,
+      initialValues: {
+        url: props.url,
+        sourceLanguageId: undefined,
+        targetLanguages: [],
+      },
+    })
+
+    type Field = { id: number; voiceType: VoiceType }
+    const targetLanguageFields = reactive<Field[]>([
+      { id: getId(), voiceType: 'Man' },
+    ])
+
+    const someTargetLanguageValid = computed(() =>
+      form.values.targetLanguages?.some((lang) =>
+        targetLanguageSchema.isValidSync(lang)
+      )
+    )
+
+    const payButtonEnabled = computed(
+      () => form.meta.value.valid && someTargetLanguageValid.value
+    )
+
+    const sourceLanguageOptions = computed(() =>
+      props.sourceLanguages.map((lang) => ({
+        value: lang.id,
+        label: lang.label,
+      }))
+    )
+
+    const availableTargetLanguages = computed(() =>
+      props.targetLanguages.filter(
+        (lang) => lang.id !== form.values.sourceLanguageId
+      )
+    )
+
+    const handleSourceLanguageChange = (newValue: string) => {
+      form.setFieldValue('sourceLanguageId', newValue)
+    }
+
+    const handleTargetLanguageVoiceTypeChange = (idx: number) => (
+      newVoiceType: VoiceType
+    ) => {
+      targetLanguageFields[idx].voiceType = newVoiceType
+    }
+
+    const handleTargetLanguageAdd = (field: Field) => () => {
+      const index = targetLanguageFields.indexOf(field)
+
+      targetLanguageFields.splice(index + 1, 0, {
+        id: getId(),
+        voiceType: 'Man',
+      })
+    }
+
+    const handleTargetLanguageRemove = (field: Field) => () => {
+      const index = targetLanguageFields.indexOf(field)
+
+      targetLanguageFields.splice(index, 1)
+      form.values.targetLanguages?.splice(index, 1)
+    }
+
+    const handleSubmit = async (event: Event) => {
+      event.preventDefault()
+      const validationResult = await form.validate()
+
+      if (validationResult.valid) {
+        props.onSubmit?.({
+          ...form.values,
+          targetLanguages: form.values.targetLanguages?.filter((lang) =>
```

Comment:
Кажется, здесь достаточно

```suggestion
          targetLanguages: form.values.targetLanguages?.filter(Boolean)
         
```